src/net/sourceforge/kolmafia/request/AdventureRequest.java

```diff
@@ -619,13 +622,34 @@ public static final String registerEncounter(final GenericRequest request) {
                 && !EncounterManager.isSpookyVHSTapeMonster(responseText, false)
                 && !EncounterManager.isMimeographEncounter(responseText)
                 && !FightRequest.edFightInProgress())) {
-          AdventureQueueDatabase.enqueue(KoLAdventure.lastVisitedLocation(), encounter);
+          AdventureQueueDatabase.enqueue(location, encounter);
         }
-      } else if (type.equals("Noncombat")) {
+      } else if (type.equals("Noncombat")
+          // Don't enqueue Lucky, hallowiener, etc. adventures.
+          && (encounterData == null
+              || encounterData.getEncounterType() == EncounterManager.EncounterType.NONE
+              || encounterData.getEncounterType() == EncounterManager.EncounterType.STOP)
+          // Special-case first two NCs in the Upper Chamber, which are superlikely.
+          // In a perfect world, Mafia would have a list of all superlikelies, but this works for
+          // now.
+          && !(AdventureSpentDatabase.getTurns(
+                      AdventureDatabase.getAdventure(AdventurePool.UPPER_CHAMBER))
+                  == 2
+              && encounter.equals("A Wheel -- How Fortunate!"))
+          && !(AdventureSpentDatabase.getTurns(
+                      AdventureDatabase.getAdventure(AdventurePool.UPPER_CHAMBER))
+                  == 5
```

Comment:
you can get past the turns spent by e.g. using professor copies to chain fights, right? So just checking the encounter name would be better (I think they are unique).

Reply:
> you optimally would have a tracking property as to whether the first one has occurred

why not do this then? If we've never encountered A Wheel, don't track the NC as it's the superlikely version, otherwise it's a normal NC? This will be wrong for adventures outside of Mafia, but will fix itself when somebody hits the NC for the first time.